From 18a6e70182f2a4dd4ea19a84eaad4ef4340831e7 Mon Sep 17 00:00:00 2001
From: Peng Zhou <benjo.zhou@rock-chips.com>
Date: Fri, 27 Oct 2017 15:17:12 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3368-android: add node for cif
 controller

rk3368-android.dtsi: increase cma-heap size
add new file for rk3368_cif_sensor.

Change-Id: I6f3fa0cd8f5a640b7d5ad87410e406bd0bf6844b
Signed-off-by: Peng Zhou <benjo.zhou@rock-chips.com>
---
 .../boot/dts/rockchip/rk3368-android.dtsi     | 16 ++++++++-
 .../boot/dts/rockchip/rk3368-cif-sensor.dtsi  | 34 +++++++++++++++++++
 2 files changed, 49 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
index 50c246379035..d648abce8508 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
@@ -40,6 +40,8 @@
  *     OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include "rk3368-cif-sensor.dtsi"
+
 / {
 	chosen {
 		bootargs = "earlycon=uart8250,mmio32,0xff690000 swiotlb=1 firmware_class.path=/system/vendor/firmware";
@@ -81,7 +83,7 @@
 		#size-cells = <0>;
 
 		cma-heap {
-			reg = <0x00000000 0x02000000>;
+			reg = <0x00000000 0x2800000>;
 		};
 
 		system-heap {
@@ -250,6 +252,18 @@
 	status = "okay";
 };
 
+&cif {
+	status = "okay";
+};
+
+&vip_mmu {
+	status = "okay";
+};
+
+&rk3368_cif_sensor {
+	status = "okay";
+};
+
 &usb_otg {
 	status = "okay";
 	clocks = <&cru SCLK_OTGPHY0>, <&cru HCLK_OTG0>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi
new file mode 100644
index 000000000000..33bde54a0431
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3368-cif-sensor.dtsi
@@ -0,0 +1,34 @@
+#include "../../../../../drivers/soc/rockchip/rk_camera_sensor_info.h"
+
+/{
+	rk3368_cif_sensor: rk3368_cif_sensor{
+		compatible = "rockchip,sensor";
+		status = "okay";
+
+		tp2825 {
+			status = "okay";
+			is_front = <0>;
+			powerdown-gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+			irq-gpios = <&gpio1 13 IRQ_TYPE_EDGE_FALLING>;
+			pwdn_active = <tp2825_PWRDN_ACTIVE>;
+			mir = <0>;
+			flash_attach = <1>;
+			flash_active = <1>;
+			resolution = <tp2825_FULL_RESOLUTION>;
+			powerup_sequence = <tp2825_PWRSEQ>;
+			orientation = <0>;
+			i2c_add = <tp2825_I2C_ADDR>;
+			i2c_chl = <3>;
+			cif_chl = <0>;
+			ad_chl = <0>; // 0 ~ 4;
+			mclk_rate = <24>;
+			rockchip,camera-module-defrect0 = <960 480 0 4 960 472>;
+			rockchip,camera-module-interface0 = "cvbs_ntsc";
+			rockchip,camera-module-defrect1 = <960 576 0 4 960 568>;
+			rockchip,camera-module-interface1 = "cvbs_pal";
+			rockchip,camera-module-defrect2 = <1280 720 8 20 1280 720>;
+			rockchip,camera-module-interface2 = "bt601_8_pp";
+			rockchip,camera-module-channel = <4 0>;
+		};
+	};
+};
-- 
2.35.3

